### ISSUE TITLE
weave tries to ping de-scaled nodes and shows failed connections

### ISSUE THREAD

**Issue by notmaxx** [weaveworks/weave#3467]
## What you expected to happen?
After AWS k8s cluster de-scaling weave stop pinging gone nodes

## What happened?
Weave on single machine still tries to establish connection with nodes that removed from the cluster. That happened only on single machines (of 10 machines total)

## How to reproduce it?
In our case we scaled cluster by scaling ASGs, did perf tests and then de-scaled it back (by de-scaling ASGs)

## Anything else we need to know?
- Cloud provider - AWS
- Configuration - terraform output generated by kops 

## Versions:
```
/home/weave # /home/weave/weave --local version
weave 2.5.0

$ docker version
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:36:34 2017
 OS/Arch:      linux/amd64

$ uname -a
Linux 4.4.115-k8s #1 SMP Thu Feb 8 15:37:40 UTC 2018 x86_64 GNU/Linux

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2", GitCommit:"5fa2db2bd46ac79e5e00a4e6ed24191080aa463b", GitTreeState:"clean", BuildDate:"2018-01-18T21:10:44Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.4", GitCommit:"9befc2b8928a9426501d3bf62f72849d5cbcd5a3", GitTreeState:"clean", BuildDate:"2017-11-20T05:17:43Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

## Weave Logs:
logs - https://gist.github.com/notmaxx/b608d0dc4238ae65b7576735ea4c95dc
status connections - https://gist.github.com/notmaxx/10fe3fb8a824853493d60820e469426d

## Network:
```
$ ip route
default via 10.177.64.1 dev eth0 
10.177.64.0/20 dev eth0  proto kernel  scope link  src 10.177.64.249 
100.96.0.0/11 dev weave  proto kernel  scope link  src 100.116.0.0 
172.17.0.0/16 dev docker0  proto kernel  scope link  src 172.17.0.1 

$ ip -4 -o addr
1: lo    inet 127.0.0.1/8 scope host lo\       valid_lft forever preferred_lft forever
2: eth0    inet 10.177.64.249/20 brd 10.177.79.255 scope global eth0\       valid_lft forever preferred_lft forever
3: docker0    inet 172.17.0.1/16 scope global docker0\       valid_lft forever preferred_lft forever
6: weave    inet 100.116.0.0/11 scope global weave\       valid_lft forever preferred_lft forever

$ sudo iptables-save
https://gist.github.com/notmaxx/de7cd4951d4c9685f82e18d6fdd09d7e
```

**Comment by notmaxx** [comment]
hi @murali-reddy yes, only single node, other nodes have nothing like this

**Comment by murali-reddy** [comment]
Root cause is same as the issue reported in https://github.com/weaveworks/weave/issues/3613

```
DEBU: 2018/11/29 09:12:49.517294 [kube-peers] Noting I plan to remove  3e:6a:4e:29:47:96
E1129 09:12:49.517445    7666 runtime.go:66] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:573
/usr/local/go/src/runtime/panic.go:502
/usr/local/go/src/runtime/hashmap_fast.go:696
/go/src/github.com/weaveworks/weave/prog/kube-utils/annotations.go:104
/go/src/github.com/weaveworks/weave/prog/kube-utils/main.go:159
/go/src/github.com/weaveworks/weave/prog/kube-utils/main.go:115
/go/src/github.com/weaveworks/weave/prog/kube-utils/main.go:236
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:209
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/shared_informer.go:556
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:203
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:203
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/shared_informer.go:548
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/shared_informer.go:546
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/shared_informer.go:390
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71
/usr/local/go/src/runtime/asm_amd64.s:2361
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 23 [running]:
github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
```

Once the panic happens, kube-utils no longer running as daemon, which results in skipping `weave forget` on dead nodes.

@notmaxx issue got fixed in https://github.com/weaveworks/weave/pull/3623, fix will be available in 2.5.2 release.
